src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsModel.cs

```diff
@@ -316,33 +312,32 @@ public override IEnumerable<IEdmVocabularyAnnotation> FindDeclaredVocabularyAnno
                 {
                     return semanticsInclude.InlineVocabularyAnnotations;
                 }
-                else
-                {
-                    Enumerable.Empty<IEdmVocabularyAnnotation>();
-                }
             }
 
             CsdlSemanticsElement semanticsElement = element as CsdlSemanticsElement;
             IEnumerable<IEdmVocabularyAnnotation> inlineAnnotations = semanticsElement != null && semanticsElement.Model == this ? semanticsElement.InlineVocabularyAnnotations : Enumerable.Empty<IEdmVocabularyAnnotation>();
 
-            List<CsdlSemanticsAnnotations> elementAnnotations;
-            string fullName = EdmUtil.FullyQualifiedName(element);
-
-            if (fullName != null && this.outOfLineAnnotations.TryGetValue(fullName, out elementAnnotations))
+            if (this.outOfLineAnnotations.Count > 0)
             {
-                List<IEdmVocabularyAnnotation> result = new List<IEdmVocabularyAnnotation>();
+                List<CsdlSemanticsAnnotations> elementAnnotations;
+                string fullName = semanticsElement.GetAnnotationFullQualifiedName(element);

```

Comment:
This code works under the assumption CsdlSemanticsElement instances are always the same (i.e. reused during process lifetime) and immutable, which should be the case of AGS. Not sure there are other scenarios, edge cases to cover?